matrix_sdk/src/client.rs

```diff
@@ -1220,9 +1220,13 @@ impl Client {
     ///
     /// * `sync_settings` - Settings for the sync call.
     #[instrument]
-    pub async fn sync(&self, sync_settings: SyncSettings) -> Result<sync_events::Response> {
+    pub async fn sync(
+        &self,
+        sync_settings: SyncSettings,
+        filter: Option<sync_events::Filter>,
```

Comment:
This needs docs.

Also since we already have a `SyncSettings` object will probably want the filter to live there instead of adding another argument to this method.